Automattic/wp-calypso
# build: Skip `tsc` on docker images

#### Background

We have a `postinstall` job that runs `build-packages`, which runs `tsc` on all `packages/*`.

However, the result of that compilation is ignored in CI, as we rely on Webpack to transpile `.ts` packages before bundling them. That `tsc` command only makes sense in development, as it generates the types used by IDEs to provide a better TypeScript experience.

As such, time ago we introduced a `SKIP_TSC` env var to skip that step in CI. However, it is not working as expected:

- When running builds in TeamCity, it still seems to run `tsc`. Note how there is an internval of ~50 seconds before lerna starts, a sign that `tsc` is running:

![image](https://user-images.githubusercontent.com/975703/122899856-7c38bb00-d34c-11eb-8f3f-af68db2d245e.png)

- When building the final docker image for production, it fails because it uses `/bin/sh` instead of `/bin/bash`. This results in `tsc` being run (gap of ~40s)

![image](https://user-images.githubusercontent.com/975703/122899806-717e2600-d34c-11eb-825a-08c393476c31.png)


#### Changes proposed in this Pull Request

* Change `[[` to `[` to avoid any bash specific syntax.
* Hardcode env var `SKIP_TSC` in TeamCity build plans

#### Testing instructions

* Check the logs for the `Docker image` build and verify it doesn run `tsc` (i.e `lerna` starts right away)

![image](https://user-images.githubusercontent.com/975703/122911730-e1de7480-d357-11eb-91d9-0dbd59e5cba2.png)
